tests/fake_http_server/fake_test_server.py

```diff
@@ -160,28 +169,67 @@ def do_PUT(self):
         self.do_POST()
 
 
-class ReUseHTTPServer(HTTPServer):
-    def __init__(self, addr, meta_path, target_path, fail_injector=None):
+class FakeTestServer(socketserver.ThreadingMixIn, HTTPServer):
+    def __init__(self, addr, meta_path, target_path, srcdir=None, fail_injector=None):
         super(HTTPServer, self).__init__(server_address=addr, RequestHandlerClass=Handler)
         self.meta_path = meta_path
-        self.target_path = target_path
+        if target_path is not None:
+            self.target_path = target_path
+        elif meta_path is not None:
+            self.target_path = path.join(meta_path, 'repo/image/targets')
+        else:
+            self.target_path = None
         self.fail_injector = fail_injector
+        self.srcdir = srcdir if srcdir is not None else os.getcwd()
 
     def server_bind(self):
         self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
         HTTPServer.server_bind(self)
 
 
+class FakeTestServerBackground:
+
+    def __init__(self, meta_path, target_path=None, srcdir=None, port=0):
+        if srcdir is None:
+            srcdir = os.getcwd()
+        self._httpd = FakeTestServer(addr=('', port), meta_path=meta_path,
+                                     target_path=target_path, srcdir=srcdir)
+        self._server_process = self.__class__.Process(target=self._httpd.serve_forever)
+
+    class Process(multiprocessing.Process):
+        def run(self):
+            with open(os.devnull, 'w') as devnull_fd:
+                with contextlib.redirect_stdout(devnull_fd),\
+                        contextlib.redirect_stderr(devnull_fd):
+                    super(multiprocessing.Process, self).run()
+
+    @property
+    def port(self):
+        return self._httpd.server_port
+
+    def __enter__(self):
+        self._server_process.start()
+        logger.debug("Uptane server running and listening: {}".format(self.port))
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._server_process.terminate()
+        self._server_process.join(timeout=10)
+        logger.debug("Uptane server has been stopped")
+
+
```

Comment:
It looks quite similar to UptaneTestBackend, perhaps it makes sense to bring it to common 'denominator' ?